src/blocks/subscribe/index.php

```diff
@@ -502,15 +502,9 @@ function process_form() {
 			'email'    => $email,
 			'metadata' => $metadata,
 		],
-		$lists,
-		true // Async.
+		$lists
 	);
 
-	// The async subscription strategy returns true.
-	if ( true === $result ) {
-		$result = [];
-	}
-
 	if ( \is_wp_error( $result ) ) {
 		return send_form_response( $result );
 	}
```

Comment:
I have a general question about the behavior of this process_form function here. In it, we first create wp user, and then attempt to create a contact in the ESP. In the event the contact fails to be added, what happens is we have a wp account that has no corresponding ESP contact data. 

Is this the expected behavior? If not, should we flip this around and attempt to create the contact first, only creating a wp user if that is successful. This would prevent us from adding wp accounts for invalid email addresses that aren't accepted by the ESP (test@invalid.com for example)

Reply:
Good idea! I agree that if adding the contact to the ESP fails, the WP users should not be added. The order of events was intentionally switched in https://github.com/Automattic/newspack-newsletters/pull/1250, but the context there is segmentation. @dkoo – can you weigh in here? Can the segmentation fix from #1250 be achieved while also implementing the change suggested above?

Reply:
I think this might be a subjective thing, but IMO creating the WP account is more important than the contact data sync. Even if the initial data sync to the ESP fails, if the reader performs another action that would trigger a contact sync (like completing a new transaction, having a subscription renewal or status update, signing up for another newsletter list, unsubscribing from a newsletter list, logging into another browser session etc.) the contact data will be created at the first successful sync and will reflect the reader's most recent state. In other words, the WP reader account should be the source of truth, not the ESP data, since we can always correct ESP data after the fact, but if the WP account isn't created in the first place, the reader's activity won't be persisted in future sessions.